src/controllers/simplified-account/settings/card-payments/google-pay.controller.js

```diff
@@ -0,0 +1,26 @@
+const { response } = require('@utils/response')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const { onOrOffToBool } = require('@utils/on-or-off')
+const paths = require('@root/paths')
+const { updateGooglePay } = require('@services/card-payments.service')
+
+function get (req, res) {
+  const googlePay = req.account?.allowGooglePay
+  response(req, res, 'simplified-account/settings/card-payments/google-pay', {
+    currentState: googlePay ? 'on' : 'off',
+    backLink: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.cardPayments.index, req.service.externalId, req.account.type)
+  })
+}
+
+async function post (req, res) {
+  const userPreference = onOrOffToBool(req.body.googlePay)
```

Comment:
as it stands, this will throw an unhandled error and cause the error handler middleware to return a 500 and show the 'problem with the platform' page if someone messes with the DOM prior to submitting the form.

ideally, we'd just handle this (i.e. return the user to the form) without showing an error page (or perhaps show some validation error)

Reply:
Yes, I think this is the same issue Dom mentioned above, just highlighting a different place in the code. Happy to add that, but we aren't validating for users messing with the DOM in other controllers, e.g. Email notifications, card types. If we want this then we should probably add a ticket to sweep up those other instances too. @oswaldquek any preference?

Reply:
I've added validation, but we (the team) should probably talk about what our expectations are here.